### DISCUSSION TITLE
Is there a way to get a locale-correct `orderby()`?

### DISCUSSION THREAD

**Discussion by agladysh** [uwdata/arquero#366] [Q&A] [answered]
Problem:

```csv
çilek
erik
şeftali
armut
üzüm
```

By default, `orderby()` gives incorrect (for `tr-TR` locale) sorting order:

```js
aq.fromCSV(csv, { header: false }).orderby('col1').toCSV()
```

```csv
armut
erik
çilek
üzüm
şeftali
```

Correct order would be:

```csv
armut
çilek
erik
şeftali
üzüm
```

One can achieve that by using `String.localeCompare()` in the comparator:

```js
(lhs, rhs) => lhs.localeCompare(rhs, 'tr-TR');
```

Is there a way to feed this comparator to the Arquero somehow?

**Comment by jheer** [comment]
There is not a mechanism for this at present. Currently Arquero relies on standard JS comparison operators. I'll open a feature request issue for this.

**Comment by jheer** [comment, marked as answer]
[Arquero version 7.1.0](https://github.com/uwdata/arquero/releases/tag/v7.1.0) now includes a [`collate` helper function](https://idl.uw.edu/arquero/api/#collate) to support custom string orders. You can use it in your example above like so:

```js
aq.fromCSV(csv, { header: false }).orderby(aq.collate('col1', 'tr-TR')).toCSV()
```

The collate method accepts either arguments compatible with the [Intl.Collator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator) constructor (such as a locale string) or a two-argument custom comparator function.

**Reply by agladysh** [reply]
Wow! Thank you! This helps a lot!
